R/ari_spin.R

```diff
@@ -57,115 +49,107 @@
-                     service = ifelse(have_polly(), "amazon", "google"),
+                     tts_engine = text2speech::tts,
+                     tts_engine_args = list(service = "coqui",
+                                            voice = NULL,
+                                            model_name = "tacotron2-DDC_ph",
+                                            vocoder_name = "ljspeech/univnet"),
+                     tts_engine_auth = text2speech::tts_auth,
                      subtitles = FALSE,
                      duration = NULL,
-                     tts_args = NULL,
                      key_or_json_file = NULL,
-                     ...) {
+                     verbose = FALSE,
+                     cleanup = TRUE) {
   # Check for ffmpeg
   ffmpeg_exec()
-
   # Argument checks
-  auth <- text2speech::tts_auth(
-    service = service,
+  auth <- tts_engine_auth(
+    service = tts_engine_args$service,
     key_or_json_file = key_or_json_file
   )
   if (!auth) {
     stop(paste0(
       "It appears you're not authenticated with ",
-      service, ". Make sure you've ",
+      tts_engine_args$service, ". Make sure you've ",
       "set the appropriate environmental variables ",
       "before you proceed."
     ))
   }
-
+  # Create file path to output
   stopifnot(length(images) > 0)
   images <- normalizePath(images)
   output_dir <- normalizePath(dirname(output))
-
+  # Paragraphs
   if (length(paragraphs) == 1) {
     if (file.exists(paragraphs)) {
       paragraphs <- readLines(paragraphs, warn = FALSE)
       paragraphs <- paragraphs[!paragraphs %in% ""]
     }
   }
-
+  # Paragraphs: Check for semicolons
   semi_colon <- trimws(paragraphs) == ";"
   if (any(semi_colon)) {
     warning(paste0(
       "Some paragraphs are simply a semicolon - ",
       "likely needs to be replaced or slide removed!"
     ))
   }
+  # Check for arguments
   stopifnot(
     length(paragraphs) > 0,
     identical(length(images), length(paragraphs)),
     all(file.exists(images)),
     dir.exists(output_dir)
   )
-  # End of Argument checks
 
   # Setup objects to populate in for-loop with tts()
   wave_objects <- vector(mode = "list", length = length(paragraphs))
-  par_along <- seq_along(paragraphs)
+  paragraphs_along <- seq_along(paragraphs)
   ideal_duration <- rep(NA, length(paragraphs))
 
-  pb <- progress_bar$new(
-    format = "Fetching Narration [:bar] :percent",
-    total = length(par_along)
-  )
   # Iterate through arguments used in tts()
-  for (i in par_along) {
-    args <- tts_args
-    args$text <- paragraphs[i]
-    args$voice <- voice
-    args$service <- service
+  for (ii in paragraphs_along) {
+    args <- tts_engine_args
+    args$text <- paragraphs[ii]
     args$bind_audio <- TRUE
     # coqui+ari doesn't work with mp3
-    if (service == "coqui") {
+    if (tts_engine_args$service == "coqui") {
       args$output_format <- "wav"
+      args$voice <- NULL
     }
```

Comment:
Considering we are implementing these options in `coqui_args()`, remove the following:

```
if (tts_engine_args$service == "coqui") {
      args$output_format <- "wav"
      args$voice <- NULL
}
```